snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/DeviceUtils.kt

```diff
@@ -256,43 +210,24 @@ fun configToDeviceSpec(config: ComposePreviewSnapshotConfig): DeviceSpec? {
   return when {
     devicePreviewString?.name != null -> KNOWN_DEVICE_SPECS[devicePreviewString.name]
     devicePreviewString?.id != null -> KNOWN_DEVICE_SPECS[devicePreviewString.id]
-    else -> null
-  }
-}
-
-fun configToDimensionSpec(
-  config: ComposePreviewSnapshotConfig,
-): DimensionSpec {
-  val devicePreviewString = parseDevicePreviewString(config.device)
-  val densityPpi = devicePreviewString?.dpi?.let { it / MDPI_THRESHOLD }
-  val dimensionSpec: DimensionSpec? = when {
-    devicePreviewString?.name != null -> KNOWN_DEVICE_SPECS[devicePreviewString.name]?.dimensionSpec
-    devicePreviewString?.id != null -> KNOWN_DEVICE_SPECS[devicePreviewString.id]?.dimensionSpec
     else -> {
-      var widthDp = devicePreviewString?.widthDp
-      var heightDp = devicePreviewString?.heightDp
-      // Flip to match orientation
+      var widthDp = config.widthDp ?: devicePreviewString?.widthDp
+      var heightDp = config.heightDp ?: devicePreviewString?.heightDp
+
       if (devicePreviewString?.orientation == "landscape") {
-        heightDp = devicePreviewString.widthDp
-        widthDp = devicePreviewString.heightDp
+        val height = heightDp
+        heightDp = widthDp
+        widthDp = height
       }
-      DimensionSpec(
-        widthDp = widthDp,
-        heightDp = heightDp,
-        densityPpi = densityPpi,
-        scalingFactor = densityPpi?.let(DeviceSpec::getClosestDensityScalingFactor)
+
+      DeviceSpec(
+        heightDp = heightDp ?: -1,
```

Comment:
Is it possible to get rid of these -1 representations everywhere and instead use nil?